paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -471,6 +468,131 @@ public boolean caseSensitive() {
         return false;
     }
 
+    @Override
+    public void repairCatalog() {
+        List<String> databases = listDatabases(new Path(warehouse));
+        for (String database : databases) {
+            repairDatabase(database);
+        }
+    }
+
+    @Override
+    public void repairDatabase(String databaseName) {
+        repairHmsDatabase(databaseName);
+
+        List<String> tables = listTables(databaseName, true);
+
+        CompletableFuture<Void> allOf =
+                CompletableFuture.allOf(
+                        tables.stream()
+                                .map(table -> Identifier.create(databaseName, table))
+                                .map(
+                                        identifier ->
+                                                CompletableFuture.runAsync(
+                                                        () -> {
+                                                            try {
+                                                                repairTable(identifier);
+                                                            } catch (TableNotExistException e) {
+                                                                LOG.error(
+                                                                        "Table {} does not exist in the paimon.",
+                                                                        identifier.getFullName());
+                                                                // ignore
+                                                            }
+                                                        },
+                                                        COMMON_IO_FORK_JOIN_POOL))
+                                .toArray(CompletableFuture[]::new));
+        allOf.join();
+    }
+
+    private void repairHmsDatabase(String databaseName) {
+        checkNotSystemDatabase(databaseName);
+        try {
+            Database database = client.getDatabase(databaseName);
+            Path newPath = newDatabasePath(databaseName);
+            if (!locationHelper.getDatabaseLocation(database).equals(newPath.toUri().toString())) {
```

Comment:
If the database already exists, we will use the location set in it. We cannot change the user's location.

Please add tests to cover this.

Reply:
done.